sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/implementation/converter/AzureAmqpProxyOptionsConverter.java

```diff
@@ -32,18 +33,25 @@ public ProxyOptions convert(ProxyOptionsAware.Proxy proxy) {
             LOGGER.debug("Proxy hostname or port is not set.");
             return null;
         }
-        ProxyAuthenticationType authenticationType;
-        switch (proxy.getAuthenticationType()) {
-            case "basic":
-                authenticationType = ProxyAuthenticationType.BASIC;
-                break;
-            case "digest":
-                authenticationType = ProxyAuthenticationType.DIGEST;
-                break;
-            default:
-                authenticationType = ProxyAuthenticationType.NONE;
+        ProxyAuthenticationType authenticationType = ProxyAuthenticationType.NONE;
+        if (proxy instanceof ProxyOptionsAware.AmqpProxy) {
+            ProxyOptionsAware.AmqpProxy amqpProxy = (ProxyOptionsAware.AmqpProxy) proxy;
+            if (amqpProxy.getAuthenticationType() != null) {
+                switch (amqpProxy.getAuthenticationType().toLowerCase(Locale.ROOT)) {
+                    case "basic":
+                        authenticationType = ProxyAuthenticationType.BASIC;
+                        break;
+                    case "digest":
+                        authenticationType = ProxyAuthenticationType.DIGEST;
+                        break;
+                    default:
+                }
+            }
         }
         Proxy.Type type;
+        if (proxy.getType() == null) {
```

Comment:
do we need to add a "socks5" case here?

Reply:
No for now, the underline amqp options doesn't have a socks4 or sock5, they only have one socks